crates/ruff/tests/lint.rs

```diff
@@ -1168,3 +1168,99 @@ def func():
 
     Ok(())
 }
+
+/// Per-file selects via ! negation in per-file-ignores
+#[test]
+fn negated_per_file_ignores() -> Result<()> {
+    let tempdir = TempDir::new()?;
+    let ruff_toml = tempdir.path().join("ruff.toml");
+    fs::write(
+        &ruff_toml,
+        r#"
+[lint.per-file-ignores]
+"!selected.py" = ["F"]
```

Comment:
FWIW my preference is to use the test rules instead of real rules, they're less likely to change and it avoids mixing concerns — most tests don't use them yet because they're relatively new. If it's a pain this is fine though.

Reply:
Makes sense, will do.